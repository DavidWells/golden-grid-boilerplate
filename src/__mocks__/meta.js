// eslint-disable-next-line import/prefer-default-export
export const meta = {
  lang: 'en',
  path: './',
  charset: 'UTF-8',
  title: 'Golden canon grid website boilerplate',
  metaDesc: 'Georgina Alson portfolio golden canon grid website boilerplate created with css grid',
  template: 'base-template',
  // String value
  home: {
    name: 'Georgina',
    surname: 'Alson',
    profession: 'YUNG MODEL 2020',
    menuLink_1: {
      link: '/',
      name: 'CLIENTS'
    },
    menuLink_2: {
      link: '/',
      name: 'NEWS'
    },
    profileLink: {
      link: '/',
      name: 'VIEW PROFILE'
    }
  },
  // Navigation menu
  navigation: [
    {
      title: 'Home',
      link: '/index.html'
    },
    {
      title: 'Profile',
      link: '/profile.html'
    },
    {
      title: 'Contact',
      link: '/contact.html'
    },
    {
      title: 'Client',
      link: '/client.html'
    },
    {
      title: 'News',
      link: '/news.html'
    },
    {
      title: 'Single Blog',
      link: '/single-blog.html'
    },
    {
      title: 'About',
      link: '/about.html'
    }
  ],
  social: [
    {
      id: '#facebook',
      title: 'facebook'
    },
    {
      id: '#instagram',
      title: 'instagram'
    },
    {
      id: '#youtube',
      title: 'youtube'
    },
    {
      id: '#twitter',
      title: 'twitter'
    }
  ]
};
