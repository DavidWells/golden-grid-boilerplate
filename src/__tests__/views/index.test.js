import 'cross-fetch/polyfill';
import pug from 'pug';
import path from 'path';
import chalk from 'chalk';
import * as mocks from '../../__mocks__';

const renderPug = (data) => pug.renderFile(path.resolve(__dirname, '../../views/index.pug'), data);

describe(`Testing view: [${chalk.yellowBright('index')}]`, () => {
  it('Should render and match snapshot', () => {
    expect(renderPug(mocks.meta)).toMatchSnapshot();
  });
});
