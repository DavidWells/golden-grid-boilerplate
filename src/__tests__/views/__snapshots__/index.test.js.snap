// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing view: [[93mindex[39m] Should render and match snapshot 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head><meta charset=\\"UTF-8\\"><title>Home page - Golden canon grid website boilerplate</title><meta name=\\"description\\" content=\\"Georgina Alson portfolio golden canon grid website boilerplate created with css grid\\"><meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0, shrink-to-fit=no\\"><meta name=\\"theme-color\\" content=\\"#ffffff\\"><meta http-equiv=\\"Content-Security-Policy\\" content=\\"font-src 'self' 'unsafe-inline' 'unsafe-eval' https://fonts.gstatic.com;\\"><meta http-equiv=\\"Content-Security-Policy\\" content=\\"script-src 'self' 'unsafe-inline' 'unsafe-eval' http://www.google.com https://fonts.googleapis.com;\\"><meta http-equiv=\\"Content-Security-Policy\\" content=\\"style-src 'self' 'unsafe-inline' 'unsafe-eval' https://fonts.googleapis.com;\\"><!-- Microsoft --><meta name=\\"msapplication-TileColor\\" content=\\"#ffffff\\"><meta name=\\"msapplication-TileImage\\" content=\\"./favicon-144x144.png\\"><!-- Apple --><link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"./favicon-180x180.png\\"><link rel=\\"preconnect\\" href=\\"https://fonts.gstatic.com/\\" crossorigin><link rel=\\"manifest\\" href=\\"./manifest.json\\"><link rel=\\"icon\\" type=\\"image/png\\" href=\\"./favicon-32x32.png\\" sizes=\\"32x32\\"><link rel=\\"icon\\" type=\\"image/png\\" href=\\"./favicon-16x16.png\\" sizes=\\"16x16\\"><link href=\\"favicon.ico\\" rel=\\"shortcut icon\\" type=\\"image/x-icon\\"><base href=\\"/\\"></head><body><aside class=\\"is-hidden\\" id=\\"sidebar\\"><section class=\\"sidebar-content\\"><nav class=\\"menu\\"><ul class=\\"menu-inner\\"><li class=\\"menu-item\\"><a class=\\"menu-link is-active\\" href=\\"/index.html\\">Home</a></li><li class=\\"menu-item\\"><a class=\\"menu-link\\" href=\\"/profile.html\\">Profile</a></li><li class=\\"menu-item\\"><a class=\\"menu-link\\" href=\\"/contact.html\\">Contact</a></li><li class=\\"menu-item\\"><a class=\\"menu-link\\" href=\\"/client.html\\">Client</a></li><li class=\\"menu-item\\"><a class=\\"menu-link\\" href=\\"/news.html\\">News</a></li><li class=\\"menu-item\\"><a class=\\"menu-link\\" href=\\"/single-blog.html\\">Single Blog</a></li><li class=\\"menu-item\\"><a class=\\"menu-link\\" href=\\"/about.html\\">About</a></li></ul></nav></section><section id=\\"sidebar-mask\\"></section></aside><script>// Handles active page displaying, for demonstration purposes
function setActiveSelector(selectors, id) {
  if (selectors && selectors.length) {
    let el = selectors.length;

    while (el--) {
      console.log(selectors[el].innerText)
      if (selectors[el].innerText === \`0\${id}\`) {
        selectors[el].classList.add('active');
      } else {
        selectors[el].classList.remove('active');
      }
    }
  }
}
// Handles page changes, for demonstration purposes
function setCurrentPage(id) {
  const indicator = document.getElementById('home-page-indicator');
  const selectors = document.getElementsByClassName('pages-selector-child');

  if (indicator) {
    indicator.innerHTML= \`<h2>0\${id}</h2>\`
  }

  setActiveSelector(selectors, id);
}
</script><main class=\\"wrapper\\"><article id=\\"home-content\\"><div class=\\"navigation container__flex-row\\"><button class=\\"icon-btn ripple\\" id=\\"sidebar-btn\\"><svg class=\\"menu\\"><use href=\\"#menu\\"></use></svg></button><a class=\\"text-link\\" href=\\"/\\">CLIENTS</a><a class=\\"text-link\\" href=\\"/\\">NEWS</a></div><div class=\\"search container__flex-col\\"><button class=\\"icon-btn ripple\\"><svg class=\\"search\\"><use href=\\"#search\\"></use></svg></button></div><div class=\\"social container__flex-col\\"><button class=\\"icon-btn ripple\\"><svg class=\\"facebook\\"><use href=\\"#facebook\\"></use></svg></button><button class=\\"icon-btn ripple\\"><svg class=\\"instagram\\"><use href=\\"#instagram\\"></use></svg></button><button class=\\"icon-btn ripple\\"><svg class=\\"youtube\\"><use href=\\"#youtube\\"></use></svg></button><button class=\\"icon-btn ripple\\"><svg class=\\"twitter\\"><use href=\\"#twitter\\"></use></svg></button></div><div class=\\"name no-select\\"><h1>Georgina</h1><h1>Alson</h1></div><div class=\\"profession container__flex-col no-select\\"><h3>YUNG MODEL 2020</h3></div><div class=\\"profile container__flex-col\\"><a class=\\"text-link\\" href=\\"/\\">VIEW PROFILE</a></div><div class=\\"page-number container__flex-row no-select\\" id=\\"home-page-indicator\\"><h2>02</h2></div><div class=\\"page-number-line container\\"><svg class=\\"vertical-line\\"><use href=\\"#vertical-line\\"></use></svg></div><div class=\\"page-selector\\"><ul><li class=\\"pages-selector-child\\"><button class=\\"text-btn\\" onClick=\\"setCurrentPage(01)\\" value=\\"01\\">01</button><svg class=\\"horizontal-line\\"><use href=\\"#horizontal-line\\"></use></svg></li><li class=\\"pages-selector-child\\"><button class=\\"text-btn\\" onClick=\\"setCurrentPage(02)\\" value=\\"02\\">02</button><svg class=\\"horizontal-line\\"><use href=\\"#horizontal-line\\"></use></svg></li><li class=\\"pages-selector-child\\"><button class=\\"text-btn\\" onClick=\\"setCurrentPage(03)\\" value=\\"03\\">03</button><svg class=\\"horizontal-line\\"><use href=\\"#horizontal-line\\"></use></svg></li><li class=\\"pages-selector-child\\"><button class=\\"text-btn\\" onClick=\\"setCurrentPage(04)\\" value=\\"04\\">04</button><svg class=\\"horizontal-line\\"><use href=\\"#horizontal-line\\"></use></svg></li><li class=\\"pages-selector-child\\"><button class=\\"text-btn\\" onClick=\\"setCurrentPage(05)\\" value=\\"05\\">05</button><svg class=\\"horizontal-line\\"><use href=\\"#horizontal-line\\"></use></svg></li></ul></div><div class=\\"play container__flex-col\\"><button class=\\"icon-btn ripple\\"><svg class=\\"play\\"><use href=\\"#play\\"></use></svg></button></div><div class=\\"logo container__flex-col\\"><a class=\\"image-link\\" href=\\"/\\"><img src=\\"/assets/img/logo.png\\" alt=\\"logo\\"></a></div><div class=\\"bg-block-1\\"></div><div class=\\"bg-block-2\\"></div></article><!--article#home-clients--><!--  h2 client--><!--article#home-news--><!--  h2 news--></main></body><noscript><p>You need to enable JavaScript to run this app.</p></noscript></html>"
`;
